### PR TITLE
Add rustfmt rules to the project.

### PR DESCRIPTION
**Pull request by xgreenx**
Added rustfmt rules. I used my favorite rules but we can discuss your wishes=)
It requires usage of nightly rustfmt because it contains many new configurations.
Formatted the whole project, so it will make ongoing PR harder(you need to fix conflicts)
Added CI to check the fmt.

### CODE REVIEW

**Comment by xgreenx** [comment]
You can check what the code looks like and decide whether it is okay or not=)
And maybe better to merge other PR first before this one.

File: .github/workflows/ci.yml
Comment (Voxelot) [inline .github/workflows/ci.yml:23]: Do we need job level concurrency settings here? Seems like this is superseded by the workflow-level concurrency setting on line 12.

**Comment by xgreenx** [comment]
We actually can remove it because it is very fas check(~10 seconds) =)

File: .github/workflows/ci.yml
Comment (Voxelot) [inline .github/workflows/ci.yml:92]: we should make this job require rustfmt as a pre-requisite, otherwise CI will continue even if the formatter checks fail.

File: .rustfmt.toml
Comment (vlopes11) [inline .rustfmt.toml:31]: mixed feelings with this one too since we will end with more lines of code

File: .rustfmt.toml
Comment (vlopes11) [inline .rustfmt.toml:36]: Ditto; is it desirable to enforce more LOC?

File: .rustfmt.toml
Comment (vlopes11) [inline .rustfmt.toml:40]: This one is very good

File: .rustfmt.toml
Comment (vlopes11) [inline .rustfmt.toml:49]: Also another cool improvement!

**Review by vlopes11 — COMMENTED** [review]
I think its a good thing to have a fmt standard. But then if we are diverging from the default of rustfmt, we might want to enforce that to all our repos so we don't end up having different styles across the org

File: .rustfmt.toml
Comment (xgreenx) [reply .rustfmt.toml:2]: Of course, we can do that, but if in the future Rust team decides on default values, it will affect us=)

File: .rustfmt.toml
Comment (xgreenx) [reply .rustfmt.toml:10]: Many fields from here are unstable=) It is why it is nightly rust fmt

File: .rustfmt.toml
Comment (xgreenx) [reply .rustfmt.toml:21]: Example from our codebase("Crate"):
изображение

Example from our codebase("Module"):
изображение


I'm okay with both variants but prefer more "Crate" because I can easy collapse imports from one crate or add a `cfg` flag for whole crate=)

File: .rustfmt.toml
Comment (xgreenx) [reply .rustfmt.toml:31]: I think the number of lines is not a problem. The readability of the code is more important=) With that flag each block of code will have its idents.

File: .rustfmt.toml
Comment (xgreenx) [reply .rustfmt.toml:36]: Again, better to write readable code(for reviewers) than try to put everything in one line with `if a > b { 24 } else { 42 }`. IDE will collapse it for you in code during navigation, so you will not see it

File: .rustfmt.toml
Comment (adlerjohn) [reply .rustfmt.toml:10]: Hmmm. Does this require users to switch between `nightly` and `stable` to run `cargo fmt`? The codebase itself builds with `stable`.

Comment (xgreenx) [reply .rustfmt.toml:10]: They need to do `cargo +nightly fmt`. But all IDE support a separate channel for rustfmt.
изображение

File: .rustfmt.toml
Comment (xgreenx) [reply .rustfmt.toml:1]: I'm using only a laptop=) The default value is 120. I didn't try 100 by I know that with 90, it always fits into GitHub split mode review=)

**Comment by xgreenx** [comment]
> I think its a good thing to have a fmt standard. But then if we are diverging from the default of rustfmt, we might want to enforce that to all our repos so we don't end up having different styles across the org

Yea, if we are okay to apply rustfmt for `fuel-core` then better to do the same for all rust repos=)

**Comment by adlerjohn** [comment]
If this requires an org-wide change then it should have org-wide buy-in. Might be time to follow-up on https://github.com/FuelLabs/rfcs/pull/8

File: .github/workflows/ci.yml
Comment (Voxelot) [reply .github/workflows/ci.yml:89]: Specifically the "needs" section for `cargo-verifications` should include `rustfmt`

File: .rustfmt.toml
Comment (Voxelot) [reply .rustfmt.toml:21]: the cfg flag can go both ways. For example, it is also likely that a cfg flag will affect a subset of imports within a crate, rather than the entire crate. 


```
use crate::{ThingA, ThingB}
#[cfg(feature = "feature-c")]
use crate::ThingC
```

Will rustfmt avoid collapsing the import if we have a cfg flag on a specific set of imports from a crate?

Comment (xgreenx) [reply .rustfmt.toml:21]: Yes, it will not touch `cfg` imports. Also, if you have an "enter" between to import section, it will not merge them into one, but it will collapse and sort all inside each section(the same for `cfg`).

File: .rustfmt.toml
Comment (Voxelot) [reply .rustfmt.toml:6]: Just to iterate on the reviewability point a little more, more LOC seems like it would make diffs in github easier to follow, as the changes would be more specific and isolated.

Comment (Dentosal) [reply .rustfmt.toml:6]: I really dislike this one, especially for nested match blocks and single-expression closures. It hurts readibility quite a bit, since I'm accustomed to block having multiple expressions.

Comment (xgreenx) [reply .rustfmt.toml:6]: Better to not have nested match blocks=D It will force you to solve them

Comment (Voxelot) [reply .rustfmt.toml:6]: seems like this one isn't so popular, maybe we should revert to default here

File: .rustfmt.toml
Comment (leviathanbeak) [reply .rustfmt.toml:1]: the default value is 100 -> https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#max_width

**Comment by leviathanbeak** [comment]
Reviewed all the config options, apart from maybe `imports_granularity = "Crate/Module"` I don't see much use in enforcing other ones, that is, diverging from the default values.

**Review by Voxelot — DISMISSED** [review]
lgtm, but since this is a big change plz wait for at least one other approval before merging.

**Review by Dentosal — DISMISSED** [review]
lgtm as well

**Comment by Voxelot** [comment]
@adlerjohn we need to update the ci requirements on the repo to merge this pr

**Comment by Voxelot** [comment]
@xgreenx can you also update the `CONTRIBUTING.md` with the updated instructions for running the nightly formatter?

**Review by Voxelot — APPROVED** [review]

**Review by Voxelot — APPROVED** [review]

**Pull request merged**
